packages/core/src/utils.ts

```diff
@@ -646,6 +646,25 @@ export function isPopulated<T extends { length: number } | object>(arg: CanBePop
   );
 }
 
+/**
+ * Returns an array with trailing empty elements removed.
+ * For example, [1, 2, 3, null, undefined, ''] becomes [1, 2, 3].
+ * This is useful for FHIR arrays, which by default must maintain the same length,
+ * but while editing we may want to trim trailing empty elements.
+ * @param arr - The input array.
+ * @returns The array with trailing empty elements removed.
+ */
+export function trimTrailingEmptyElements<T>(arr: T[] | undefined): T[] | undefined {
+  if (!arr) {
+    return undefined;
+  }
+  let i = arr.length - 1;
+  while (i >= 0 && isEmpty(arr[i])) {
+    i--;
+  }
+  return i > 0 ? arr.slice(0, i + 1) : undefined;
```

Comment:
Here `i` is the index of the last non-null element in the array: if it's zero we still want to return that element
```suggestion
  return i >= 0 ? arr.slice(0, i + 1) : undefined;
```

Reply:
Eek, my last minute cleanup added a bug.  Ty 🙏